*: replace loader by lightning (#2001)
pingcap/docs: how-to/maintain/backup-and-restore/mydumper-lightning.md

---

> title: Use `Mydumper` and `TiDB Lightning` for Backup and Restoration

Reviewer: Please replace all `mydumper` with Mydumper and replace all `TiDB Lightning` with TiDB Lightning.
```suggestion
title: Use Mydumper and TiDB Lightning for Backup and Restoration
```

category: how-to

> aliases: ['/docs-cn/how-to/maintain/backup-and-restore/']

Reviewer: ```suggestion
aliases: ['/docs/dev/how-to/maintain/backup-and-restore/']
```

---

> # Use `Mydumper` and `TiDB Lightning` for Data Backup and Restoration

Reviewer: ```suggestion
# Use Mydumper and TiDB Lightning for Data Backup and Restoration
```

> This document describes how to take full back up and restore the data of TiDB using Mydumper and TiDB Lightning. For incremental backup and restoration, refer to [TiDB Binlog](/reference/tidb-binlog/overview.md).

Reviewer: ```suggestion
This document describes how to perform full backup and restoration of the TiDB data using Mydumper and TiDB Lightning. For incremental backup and restoration, refer to [TiDB Binlog](/reference/tidb-binlog/overview.md).
```

Suppose that the TiDB service information is as follows:

|Name|Address|Port|User|Password|

> |----|-------|----|----|--------|

Reviewer: ```suggestion
|:----|:-------|:----|:----|:--------|
```

|TiDB|127.0.0.1|4000|root|*|

Use the following tools for data backup and restoration:

- [Mydumper](/reference/tools/mydumper.md): to export data from TiDB
- [TiDB Lightning](/reference/tools/tidb-lightning/overview.md): to import data into TiDB

> ## Full backup and restoration using `Mydumper`/`TiDB Lightning`

Reviewer: ```suggestion
## Full backup and restoration using Mydumper/TiDB Lightning
```

> `mydumper` is a powerful data backup tool. For more information, see [`maxbube/mydumper`](https://github.com/maxbube/mydumper).

Reviewer: `see` or `refer to` should be consistently used in one document.
```suggestion
`mydumper` is a powerful data backup tool. For more information, refer to [`maxbube/mydumper`](https://github.com/maxbube/mydumper).
```

Use [Mydumper](/reference/tools/mydumper.md) to export data from TiDB and use [TiDB Lightning](/reference/tools/tidb-lightning/overview.md) to import data into TiDB.

> **Note:**
>

> > It is recommended to download [Mydumper](/reference/tools/mydumper.md) from the PingCAP website, since the R&D team adapted `mydumper` for TiDB. It is not recommended to use `mysqldump` which is much slower for both backup and restoration.

Reviewer: In technical documents, it is recommended to use `because` rather than `since` or `as` to convey the casual relationship. `since` or `as` has multiple meanings.
```suggestion
> It is recommended to download [Mydumper](/reference/tools/mydumper.md) from the PingCAP website, because the R&D team has adapted `mydumper` for TiDB. It is not recommended to use `mysqldump` which is much slower for both backup and restoration.
```

> ### Best practices for full backup and restoration using `Mydumper`/`TiDB Lightning`

Reviewer: ```suggestion
### Best practices for full backup and restoration using Mydumper/TiDB Lightning
```

    ```sql
    SELECT * FROM mysql.tidb WHERE VARIABLE_NAME = 'tikv_gc_life_time';
    ```

    ```
    +-----------------------+------------------------------------------------------------------------------------------------+
    | VARIABLE_NAME         | VARIABLE_VALUE                                                                                 |
    +-----------------------+------------------------------------------------------------------------------------------------+
    | tikv_gc_life_time     | 10m0s                                                                                          |
    +-----------------------+------------------------------------------------------------------------------------------------+
    1 rows in set (0.02 sec)
    ```

    {{< copyable "sql" >}}

    ```sql
    update mysql.tidb set VARIABLE_VALUE = '720h' where VARIABLE_NAME = 'tikv_gc_life_time';
    ```

2. After running the `mydumper` command, adjust GC value of the TiDB cluster to its original value in step 1.

    {{< copyable "sql" >}}

    ```sql
    update mysql.tidb set VARIABLE_VALUE = '10m' where VARIABLE_NAME = 'tikv_gc_life_time';
    ```

## Restore data into TiDB

> To restore data into TiDB, use TiDB Lightning to import the exported data. See [TiDB Lightning Tutorial](/reference/tools/tidb-lightning/tidb-backend.md).

Reviewer: ```suggestion
To restore data into TiDB, use TiDB Lightning to import the exported data. Refer to [TiDB Lightning Tutorial](/reference/tools/tidb-lightning/tidb-backend.md).
```
